src/lib/dataset.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * This module contains helpers and definitions related to datasets.
+ */
+
+export type DataSetColumn = {
+  name: string;
+  type?: 'integer' | 'float' | 'boolean' | 'string' | 'date' | 'object';
+};
+
+export type DataSet = {
+  columns: Array<DataSetColumn>;
+  data: Array<Array<any>>;
+};
+
+export type MongoDocument = { [k: string]: any };
+export type MongoResults = Array<MongoDocument>;
+
+/**
+ * implement set union
+ * > setUnion(new Set([1, 2, 3]), new Set([2, 4, 6]))
+ * Set([1, 2, 3, 4, 6])
+ */
+function setUnion<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  const result = new Set(set1);
+  set2.forEach(val => result.add(val));
+  return result;
+}
+
+/**
+ * transform a mongo resultset (i.e. a list of json documents) into
+ * a `DataSet` structure
+ */
+export function mongoResultsToDataset(results: MongoResults): DataSet {
```

Comment:
In terms of architecture, do we want to mimic a principle similar to our translators ? With a  file defining the generic dataset API and a mongo.ts file for the specifics of how to get to it through a mongo backend ?

Reply:
Let's do that